thunder/tests/test_inplace_functionalization.py

```diff
@@ -455,3 +455,30 @@ def f(a, b):
     b_out = jitted(a, b)
     torch.testing.assert_close(b_out, b__out)
     assert b.data_ptr() == b_out.data_ptr()
+
+
+@instantiate(
+    dtypes=NOTHING,
+)
+def test_no_self_repeat_in_subsymbols(executor, device, _):
+
+    def f(a, b, c):
+        return a.add_(b, alpha=c)
+
+    def functional_f(a, b, c):
+        return a.add(b, alpha=c)
+
+    a = make_tensor((2, 2), device=device, dtype=torch.float32)
+    b = make_tensor((2, 2), device=device, dtype=torch.float32)
+    c = make_tensor((1,), device=device, dtype=torch.float32)
+
+    a_out_ref = executor.make_callable(functional_f)(a, b, c)
+
+    jitted = executor.make_callable(f)
+    a_out = jitted(a, b, c)
+    torch.testing.assert_close(a_out, a_out_ref)
+
+    traces = thunder.last_traces(jitted)
+    for t in filter(lambda t: t._provenance is not None and "Functionalize in-place ops" in t._provenance.pss, traces):
+        for bsym in filter(lambda b: b.subsymbols, t.bound_symbols):
+            assert bsym.rhs != bsym.subsymbols[0].rhs, bsym
```

Comment:
A follow-up on our discussion in https://github.com/Lightning-AI/lightning-thunder/issues/801. I did some debug runs and I confirm that that single island of `copy_` is not coming from the fusion logic.  Then, I guess, it must have come from the functionalization transform. Shall we also check that no unnecessary `prims.copy_` were introduced into the trace?